hub/core/dataset/hub_cloud_dataset.py

```diff
@@ -325,3 +326,23 @@ def _load_link_creds(self):
             self._set_org_and_name()
             self.link_creds.org_id = self.org_id
             self.link_creds.client = self.client
+
+    def _temp_write_access(self):
+        if not self.read_only or self._locked_out:
```

Comment:
if self._locked out, then shouldn't we throw a graceful error here, rather than attempting to write and raising a lower level exception?

Reply:
Locks are ignored by views anyway.